File: Python/OOPS/Inheritance.py
#Inheritance
class Parent :
  def _sum(self) :
    s=self._i+self._j
    return s
  def _diff(self) :
    if self._i>=self._j :
      d=self._i-self._j
    else :
      d=self._j-self._i
    return d

  def __init__(self, x=0, y=0) :
    self._i=x;
    self._j=y
  def display(self) :
    print("This function is adding and subtracting two numbers")
    print(f"{self._i} {self._j}")
  def AS(self) :
    print(f"{self._sum()} {self._diff()}")

class Child(Parent) :
  def _pro(self) :
    p=self.__k*self.__l
    return p
  def _quo(self) :
    if self.__k>=self.__l :
      q=self.__k/self.__l
    else :
      q=self.__l/self.__k
    return q

  def __init__(self, x=0, y=0) :
    super().__init__(x, y)
    self.__k=x
    self.__l=y
  def display(self) :
    print("This function is adding, subtracting, multiplying and dividing two numbers")
    print(f"{self.__k} {self.__l}")
  def ASMD(self) :
    print(f"{self._sum()} {self._diff()}")
    print(f"{self._pro()} {self._quo()}")

p=Parent(22.44, 33.66)
c=Child(22.44, 33.66)
p.display()
p.AS()
c.display()
c.ASMD()

print(issubclass(Child, Parent))

File: Misc/Java/OOPS/Inheritance.java
//Inheritance
class Parent
{
    protected double i, j;

    protected double sum()
    {
        double s;
        s=i+j;
        return s;
    }
    protected double diff()
    {
        double d;
        if(i>=j)
        d=i-j;
        else
        d=j-i;
        return d;
    }

    public Parent(double x, double y)
    {
        i=x;
        j=y;
    }
    public void display()
    {
        System.out.println("This function is adding and subtracting two numbers");
        System.out.println(i+" "+j);
    }
    public void AS()
    {
        System.out.println(sum()+" "+diff());
    }
}
class Child extends Parent
{
    private double k, l;

    protected double pro()
    {
        double p;
        p=k*l;
        return p;
    }
    protected double quo()
    {
        double q;
        if(k>=l)
        q=k/l;
        else
        q=l/k;
        return q;
    }

    public Child(double x, double y)
    {
        super(x, y);
        k=x;
        l=y;
    }
    public void display()
    {
        System.out.println("This function is adding, subtracting, multiplying and dividing two numbers");
        System.out.println(k+" "+l);
    }
    public void ASMD()
    {
        System.out.println(sum()+" "+diff());
        System.out.println(pro()+" "+quo());
    }
}
public class Inheritance
{
    public static void main(String[] args)
    {
        Parent p=new Parent(22.44, 33.66);
        Child c=new Child(22.44, 33.66);
        Parent ptr=new Child(22.44, 33.66);
        p.display();
        p.AS();
        c.display();
        c.ASMD();
        ptr.display();
        ptr.AS();
    }
}